### ISSUE TITLE
DATE and time part in OCI_DateToText() and OCI_DateFromText()

### ISSUE THREAD

**Issue by dbpm** [vrogier/ocilib#123]
library.c defines timeless part for DATE fields

static otext * FormatDefaultValues[OCI_FMT_COUNT] =
{
    OCI_STRING_FORMAT_DATE,

while ocilib.h defines
OCI_STRING_FORMAT_DATE              OTEXT("YYYY-MM-DD")

I suppose it's not fully correct, because the DATE fields can hold time part as well, that way time information will be lost in OCI_DateToText() and OCI_DateFromText()

I propose to implement OCI_DateToText() using two formats by default, if user did not override default format and did not specify fmt argument:

YYYY-MM-DD if the DATE value does not define time part (i.e. 00:00:00), and 
YYYY-MM-DD HH24:MI:SS if DATE value contains valuable time part 

and OCI_DateFromText() - it will be better to handle both stings - with and without timing part by default, if user did not override default DATE format and did not specify fmt value in the call.

Though this proposal is discussable, and backward compatibility has to be provided - users which do use DATE fields for date values only should not get unexpected 00:00:00 in OCI_DateToText()  results, of course

**Comment by vrogier** [comment]
Hi,

Current behavior is conformant with to_date() and to_char() SQL function behavior.
It no user provided format, it uses the defautl buit-in format and time part is omitted.
Users can either provide a format to these functions or set once a default format  (globally or on the connection object).
Proposed changes adds extra computation in these functions that must handle both legacy OCIDate and OCIDateTime struct).
Default build-in formats are just provided to make optional to pass a format to string conversion functions. (as for SQL functions).
I don't think that setting a user default format once at library startup is such a big thing.
I prefer to stick to SQL methods behavior.

Regards,

Vincent

**Comment by dbpm** [comment]
Ok, that was discussable proposal. 

I implemented task using preliminary check of DATE string value - if strlen(value) > strlen("YYYY-MM-DD") then try to convert with non-default format, but it looked ugly for me so I proposed to change default behavior.

From other hand - this example works fine

    SELECT to_char(to_date('2011-11-11', 'YYYY-MM-DD HH24:MI:SS'), 'YYYY-MM-DD HH24:MI:SS') FROM dual

to_date() is able to convert string without specified time part.

I guess that changing default date format to YYYY-MM-DD HH24:MI:SS might be more generic and loss-less. User can always truncate the 00:00:00 part
